homeassistant/components/esphome/climate.py

```diff
@@ -232,12 +268,21 @@ def hvac_action(self) -> Optional[str]:
     @esphome_state_property
     def fan_mode(self):
         """Return current fan setting."""
-        return _fan_modes.from_esphome(self._state.fan_mode)
+        if self._state.custom_fan_mode:
+            return self._state.custom_fan_mode
+        else:
+            return _fan_modes.from_esphome(self._state.fan_mode)
 
     @esphome_state_property
     def preset_mode(self):
         """Return current preset mode."""
-        return PRESET_AWAY if self._state.away else PRESET_HOME
+        if self._static_info.supports_away:
+            return PRESET_AWAY if self._state.away else PRESET_HOME
```

Comment:
`else: return` can be replaced by a basic `return`. So:
```
if:
  return
if:
  return
return
```

Reply:
Fixed, thanks